Sources/Public/BacktraceQueueFileHelper.swift

```diff
@@ -0,0 +1,98 @@
+import Foundation
+
+@objc public class BacktraceQueueFileHelper: NSObject {
+    
+    private var minimumQueueFileSizeBytes = 4096;
+    private var maxQueueFileSizeBytes = 0
+    private var breadcrumbLogDirectory: String = ""
+        
+    public init(_ breadcrumbLogDirectory: String, maxQueueFileSizeBytes: Int) {
+        super.init()
+        self.breadcrumbLogDirectory =  breadcrumbLogDirectory
+        if (maxQueueFileSizeBytes < minimumQueueFileSizeBytes) {
+            self.maxQueueFileSizeBytes = minimumQueueFileSizeBytes;
+        } else {
+            self.maxQueueFileSizeBytes = maxQueueFileSizeBytes;
+        }
+    }
+     
+    func addInfo(_ info:[String:Any]) -> Bool {
+    
+        if let text = convertInfoIntoString(info) {
+            let textBytes = Data(text.utf8)
+            if textBytes.count > 4096 {
+                BacktraceLogger.error("We should not have a breadcrumb this big, this is a bug!")
+                return false
+            }
+            
+            let fileURL = URL(fileURLWithPath: breadcrumbLogDirectory)
+            let content = try? String(contentsOf: fileURL, encoding: .utf8)
```

Comment:
Concern: performance impact of reading then write again.

Other implementation have either appended to the end of the file until the file got too big, which saves time on reading and re-writing the entire file.

Reply:
Another question: can the writing be done asynchronous? Is this implementation thread safe?